site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -47,7 +47,53 @@ private function validateDiffViewerType(string $type) {
         }
         return true;
     }
-
+/**
+        * Route for randomizing peer assignments
+        * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/ajaxRandomizePeers", methods={"POST"})
+        * @AccessControl(role="INSTRUCTOR")
+    */ 
+    public function ajaxRandomizePeers($gradeable_id){
+        $number_to_grade=1;
+       // $gradeable_id=$_POST['gradeable_id'];
+        $gradeable = $this->tryGetGradeable($gradeable_id);
+         if ($gradeable === false) {
+             $this->core->addErrorMessage('Invalid Gradeable!');
+             $this->core->redirect($this->core->buildCourseUrl());
+         }
+         $order = new GradingOrder($this->core, $gradeable, $this->core->getUser(), true);
+         $student_array= [];
+         $student_list= [];
+         $students = $this->core->getQueries()->getUsersByRegistrationSections($order->getSectionNames());
```

Comment:
I'm worried that this use of `GradingOrder` will eventually break, especially in the light of the code
```
if ($this->isPeerGrading()) {
    //Todo: What are all sections when you have peer grading?
}
```
found in `Gradeable.php`. 

I would much prefer that we just write a new database function called `getAllStudentsByRegistrationSection`.

Reply:
And two sibling functions `getAllStudentsByRotatingSection` and `getAllStudents($include_null = false)`